theano/gpuarray/tests/test_GpuMaxAndArgmax.py

```diff
@@ -0,0 +1,139 @@
+        M1 = T.tensor5()
+        M2 = T.tensor5()
+        f1 = theano.function([M1], [T.max(M1, axis=axis1), T.argmax(M1, axis=axis1)])
+        f2 = theano.function([M2], [T.max(M2, axis=axis2), T.argmax(M2, axis=axis2)])
+        test_matrix = randomTensor(1000, 100, 10, 5, 2)
+        f1(test_matrix)
+        f2(test_matrix)
+        theano1 = f1(test_matrix)
+        theano2 = f2(test_matrix)
+        ref1 = numpyMaxAndArgmax(test_matrix, axis1)
+        ref2 = numpyMaxAndArgmax(test_matrix, axis2)
+        utt.assert_allclose(ref1, ref2)
+        utt.assert_allclose(theano1, theano2)
+        utt.assert_allclose(ref1, theano1)
+
+    def test_none(self):
+        self._basic_test_tensor5(None)
+
+    def test_all_axes(self):
+        self._basic_test_tensor5((0, 1, 2, 3, 4))
+
+    def test_1_axe(self):
+        self._basic_test_tensor5(3)
+
+    def test_2_axes(self):
+        self._basic_test_tensor5((0, 3))
+
+    def test_3_axes(self):
+        self._basic_test_tensor5((0, 3, 4))
+
+    def test_4_axes(self):
+        self._basic_test_tensor5((0, 1, 2, 4))
+
+    def test_simple(self):
+        self._basic_test_tensor5(None)
+        self._basic_test_tensor5((0, 1, 2, 3, 4))
+        self._basic_test_tensor5((4, 1, 3, 2))
+
+    def test_assert_equals(self):
+        self._basic_test_assert_equals(None, (0, 1, 2, 3, 4))
+        self._basic_test_assert_equals(0, (0, 0))
+        self._basic_test_assert_equals((4, 1, 3, 2), (1, 2, 3, 4))
+        self._basic_test_assert_equals((4, 3, 2, 1, 0), None)
+        self._basic_test_assert_equals((1, 3, 4), (1, 4, 4, 1, 3, 1, 3, 4, 3, 1, 1, 3, 1, 4, 1, 4))
+
+    def test_simple_1_axis(self):
+        self._basic_test_tensor5(0)
+        self._basic_test_tensor5(1)
+        self._basic_test_tensor5(2)
+        self._basic_test_tensor5(3)
+        self._basic_test_tensor5(4)
+
+    def test_simple_2_axis(self):
+        self._basic_test_tensor5((0, 0))
+        self._basic_test_tensor5((0, 1))
+        self._basic_test_tensor5((0, 2))
+        self._basic_test_tensor5((0, 3))
+        self._basic_test_tensor5((0, 4))
+
+        self._basic_test_tensor5((1, 0))
+        self._basic_test_tensor5((1, 1))
+        self._basic_test_tensor5((1, 2))
+        self._basic_test_tensor5((1, 3))
+        self._basic_test_tensor5((1, 4))
+
+        self._basic_test_tensor5((2, 0))
+        self._basic_test_tensor5((2, 1))
+        self._basic_test_tensor5((2, 2))
+        self._basic_test_tensor5((2, 3))
+        self._basic_test_tensor5((2, 4))
+
+        self._basic_test_tensor5((3, 0))
+        self._basic_test_tensor5((3, 1))
+        self._basic_test_tensor5((3, 2))
+        self._basic_test_tensor5((3, 3))
+        self._basic_test_tensor5((3, 4))
+
+        self._basic_test_tensor5((4, 0))
+        self._basic_test_tensor5((4, 1))
+        self._basic_test_tensor5((4, 2))
+        self._basic_test_tensor5((4, 3))
+        self._basic_test_tensor5((4, 4))
```

Comment:
I don't believe that an exhaustive test like this is productive since it will just take a very long time to run.